src/QMCWaveFunctions/TrialWaveFunction.h

```diff
@@ -341,29 +347,29 @@ class TrialWaveFunction : public MPIObjectBase
   std::vector<NewTimer*> myTimers;
   std::vector<RealType> myTwist;
 
-  // helper function for extracting a list of WaveFunctionComponent from a list of TrialWaveFunction
+  /** @{
+   *  @brief helper function for extracting a list of WaveFunctionComponent from a list of TrialWaveFunction
+   */
   std::vector<WaveFunctionComponent*> extract_WFC_list(const std::vector<TrialWaveFunction*>& WF_list, int id) const;
 
-  // temporal helper function
-  template<class T>
-  static std::vector<T*> convert_ref_to_ptr_list(const std::vector<std::reference_wrapper<T>>& ref_list)
-  {
-    std::vector<T*> ptr_list;
-    ptr_list.reserve(ref_list.size());
-    for (auto& ref : ref_list)
-    ptr_list.push_back(&ref.get());
-    return ptr_list;
-  }
+  static std::vector<WaveFunctionComponent*> extract_WFC_list(const UPtrVector<TrialWaveFunction>& WF_list, int id);
 
   static std::vector<std::reference_wrapper<WaveFunctionComponent>> extract_WFC_list(
       const std::vector<std::reference_wrapper<TrialWaveFunction>>& WF_list,
       int id);
+  /** }@ */
+
+  // helper function for extrating a list of gradients from a list of TrialWaveFunction
+  std::vector<ParticleSet::ParticleGradient_t*> extract_G_list(const std::vector<TrialWaveFunction*>& wf_list) const;
```

Comment:
Having both extract_G_list and extract_g_list seems confusing.    Maybe extract_G_list_ptr and extract_g_list_ref, which would get a little more clue as to difference in function.
It looks like there is an (incompletely applied) naming convention that lowercase is for a reference vector and upper case is for a vector of pointers ???
It seems that the goal is to use references as much as possible and avoid the use of pointers (correct?)   Is there a remaining use for a vector of pointers, or is it for backwards compatibility in places?

Reply:
We will have only one in the end. Because we are at the transition period, both of them are used. To be accurate it is ref_list and ptr_list instead of list_ref and list_ptr.

Reply:
I was updating the naming and changing to reference vectors while ye's lower layer code kept the old naming and needed the ptr lists, so this was convenient but as you point out not intelligible to anyone.  The intention is to toss the ptr_lists but especially with G and L it seems like we'll at least have a few months with both.  I think the rename refactor will be fast let me check.